src/TRDH_alg.jl

```diff
@@ -263,7 +266,8 @@ function TRDH(
     hkn = h(xkn[selected])
     hkn == -Inf && error("nonsmooth term is not proper")
 
-    Δobj = fk + hk - (fkn + hkn) + max(1, abs(fk + hk)) * 10 * eps()
+    fhmax = Mmonotone > 0 ? maximum(FHobj_hist) : fk + hk
+    Δobj = fhmax - (fkn + hkn) + max(1, abs(fhmax)) * 10 * eps()
```

Comment:
I think the denominator of $\rho_k$ also changes and becomes $(f+h)_{\max} - m_k(s_k; x_k)$.

Reply:
No, in fact, the denominator is ok, but that is a little bit by chance.

Reply:
Yes, that is probably why I could not prove the convergence and get decent results... I'll rebase with my branch linked the paper.